external-crates/move/crates/move-package/src/compilation/compiled_package.rs

```diff
@@ -472,8 +473,18 @@ impl CompiledPackage {
             }
         }
 
-        // invoke the compiler
-        let mut paths = src_deps;
+        let mut paths;
+        if std::env::var("VERIFIED_SOURCE_BUILD").is_err() {
+            paths = src_deps;
+        } else {
+            let (deps_for_current_compiler, deps_for_prior_compiler) = partition_deps_by_toolchain(
+                src_deps.clone(),
+                resolution_graph.build_options.compiler_version.clone(),
+                w,
+            )?;
+            paths = deps_for_current_compiler;
+            bytecode_deps.extend(deps_for_prior_compiler.clone());
+        }
```

Comment:
Main top level change: when `VERIFIED_SOURCE_BUILD` is set, we partition dependencies based on whether they require a previous compiler. For those dependencies, we download and compile the dependencies, and add them as bytecode dependencies for the compilation of this root package.

Reply:
This allows the build to succeed, but doesn't yet work for sui test / publish, we need  to add the dependency modules `deps_compiled_units` for those to work (follow up PR).